packages/core-mobile/app/contexts/DeeplinkContext/DeeplinkContext.tsx

```diff
@@ -39,78 +39,99 @@ export const DeeplinkContextProvider = ({
   children: React.ReactNode
 }): JSX.Element => {
   const dispatch = useDispatch()
+  const store = useStore()
   const walletState = useSelector(selectWalletState)
   const isWalletActive = walletState === WalletState.ACTIVE
   const isDeveloperMode = useSelector(selectIsDeveloperMode)
   const isNotificationBlocked = useSelector(selectIsNotificationBlocked)
   const processedFeatureFlags = useSelector(selectFeatureFlags)
   const [pendingDeepLink, setPendingDeepLink] = useState<DeepLink>()
 
-  const expireDeepLink = useCallback(() => {
-    setPendingDeepLink(undefined)
-  }, [])
+  const maybeSetActiveNetwork = useCallback(
+    (data: { [p: string]: string | number | object }) => {
+      if (
+        'chainId' in data &&
+        ['string' || 'number'].includes(typeof data.chainId)
+      ) {
+        const chainId = Number(data.chainId)
+        const network = selectNetwork(chainId)(store.getState() as RootState)
+        //check if testnet should be toggled to match chainId provided in data
+        if (network && network.isTestnet !== isDeveloperMode) {
+          dispatch(toggleDeveloperMode())
+        }
+        dispatch(setActive(chainId))
+      }
+    },
+    [dispatch, isDeveloperMode, store]
+  )
 
-  const handleNotificationCallback = useCallback(
-    ({ url, accountIndex, origin, isDevMode }: NotificationCallbackProps) => {
-      const runCallback = (): void => {
-        const avalancheChainId = isDevMode
-          ? ChainId.AVALANCHE_TESTNET_ID
-          : ChainId.AVALANCHE_MAINNET_ID
+  const maybeSetActiveAccount = useCallback(
+    (data: { [p: string]: string | number | object }) => {
+      if ('accountAddress' in data && typeof data.accountAddress === 'string') {
+        const account = selectAccountByAddress(data.accountAddress)(
+          store.getState() as RootState
+        )
+        if (account) {
+          dispatch(setActiveAccountIndex(account.index))
+        }
+      }
+      if ('accountIndex' in data && typeof data.accountIndex === 'number') {
+        dispatch(setActiveAccountIndex(data.accountIndex))
+      }
+    },
+    [dispatch, store]
+  )
 
-        isDevMode !== isDeveloperMode && dispatch(toggleDeveloperMode())
-        dispatch(setActive(avalancheChainId))
-        dispatch(setActiveAccountIndex(accountIndex))
+  const handleNotificationCallback: HandleNotificationCallback = useCallback(
+    (data: { [p: string]: string | number | object } | undefined) => {
```

Comment:
since this type is used in multiple places, maybe declare the type and reuse it?

Reply:
[Extract Type NotificationData](https://github.com/ava-labs/core-mobile/pull/1659/commits/c41c6decbe9f6e75da3df7a7b9ebc5edeeac5c9d)